### PR TITLE
Ref Typings

### PR DESCRIPTION
**Pull request by NitzanWix**
## Description
Added RefInterface to the our forwardRef function. Adding this generic sets the type of the ref to the specific interface so when passing ref to a one of our components it will show a ts error when passing an incompatible interface ref.

## Changelog
Added generic typings to the forwardRef and in the asBaseComponent usage.

## Additional info
None

### CODE REVIEW

File: demo/src/screens/incubatorScreens/IncubatorSliderScreen.tsx [hunk 1]
```diff
@@ -23,10 +23,10 @@ const IncubatorSliderScreen = () => {
   const [color, setColor] = useState(COLOR);
   const [alpha, setAlpha] = useState(1);
 
-  const slider = useRef<typeof Incubator.Slider>();
-  const customSlider = useRef<typeof Incubator.Slider>();
-  const negativeSlider = useRef<typeof Incubator.Slider>();
-  const rangeSlider = useRef<typeof Incubator.Slider>();
+  const slider = useRef<Incubator.SliderRef>(null);
```
Comment (ethanshar) [inline demo/src/screens/incubatorScreens/IncubatorSliderScreen.tsx:26]: Why do we need to pass null here?

File: src/components/button/ButtonTypes.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ export enum ButtonAnimationDirection {
   right = 'right'
 }
 
+
```
Comment (ethanshar) [inline src/components/button/ButtonTypes.tsx:30]: Remove line

File: src/components/fadedScrollView/index.tsx [hunk 3]
```diff
@@ -11,6 +11,7 @@ import useScrollEnabler from '../../hooks/useScrollEnabler';
 import useScrollReached from '../../hooks/useScrollReached';
 import {forwardRef, ForwardRefInjectedProps} from '../../commons/new';
 import {ScrollView as GestureScrollView} from 'react-native-gesture-handler';
+import {ComponentStatics} from 'src/typings/common';
```
Comment (ethanshar) [inline src/components/fadedScrollView/index.tsx:14]: Fix import to be relative

Comment (ethanshar) [reply src/components/fadedScrollView/index.tsx:14]: And while here, let's reorder the imports in this file so packages imports are first and internal imports are second

File: src/components/slider/GradientSlider.tsx [hunk 4]
```diff
@@ -213,4 +213,5 @@ GradientSlider.displayName = 'GradientSlider';
 GradientSlider.types = GradientSliderTypes;
 
 // eslint-disable-next-line max-len
-export default asBaseComponent<GradientSliderProps, typeof GradientSlider>(forwardRef(asSliderGroupChild(forwardRef(GradientSlider))));
+// @ts-expect-error
```
Comment (ethanshar) [inline src/components/slider/GradientSlider.tsx:216]: Do you know why we have an error here? 
If it's too complicated we can solve it another time..

File: src/incubator/Slider/index.tsx [hunk 5]
```diff
@@ -15,7 +15,7 @@ import {
 } from './SliderPresenter';
 import Thumb from './Thumb';
 import Track from './Track';
-
+import {ComponentStatics} from 'src/typings/common';
```
Comment (ethanshar) [inline src/incubator/Slider/index.tsx:18]: Fix to relative import

**Review by ethanshar — COMMENTED** [review]
Looks great! 
Wrote few small comments..

File: demo/src/screens/incubatorScreens/IncubatorSliderScreen.tsx (hunk 1 above)
Comment (NitzanWix) [reply demo/src/screens/incubatorScreens/IncubatorSliderScreen.tsx:26]: When using it without null, the type of the ref is Ref<SliderRef | undefined> which causes an error. This error also occurs with regular react native refs. For example using React Native TextInput.
image

**Review by ethanshar — APPROVED** [review]

**Pull request merged**